ledger/simulation/resources.go

```diff
@@ -605,3 +607,696 @@ func (p *resourcePolicy) AvailableBox(app basics.AppIndex, name string, operatio
+	groupResources := struct {
+		Assets        []basics.AssetIndex
+		Apps          []basics.AppIndex
+		Accounts      []basics.Address
+		Boxes         []logic.BoxRef
+		AssetHoldings []ledgercore.AccountAsset
+		AppLocals     []ledgercore.AccountApp
+	}{
+		Apps:          []basics.AppIndex{},
+		Accounts:      []basics.Address{},
+		Boxes:         []logic.BoxRef{},
+		AssetHoldings: []ledgercore.AccountAsset{},
+		AppLocals:     []ledgercore.AccountApp{},
+	}
+
+	// Sort assets
+	for asset := range groupResourceTracker.Assets {
+		groupResources.Assets = append(groupResources.Assets, asset)
+	}
+	slices.SortFunc(groupResources.Assets, func(a, b basics.AssetIndex) int {
+		return cmp.Compare(a, b)
+	})
+
+	// Sort apps
+	for app := range groupResourceTracker.Apps {
+		groupResources.Apps = append(groupResources.Apps, app)
+	}
+	slices.SortFunc(groupResources.Apps, func(a, b basics.AppIndex) int {
+		return cmp.Compare(a, b)
+	})
+
+	// Sort accounts
+	for account := range groupResourceTracker.Accounts {
+		groupResources.Accounts = append(groupResources.Accounts, account)
+	}
+	slices.SortFunc(groupResources.Accounts, func(a, b basics.Address) int {
+		return cmp.Compare(a.GetUserAddress(), b.GetUserAddress())
+	})
+
+	// Sort boxes
+	// To sort boxes, we turn the app into a string, concat with the name, and then sort
+	for box := range groupResourceTracker.Boxes {
+		groupResources.Boxes = append(groupResources.Boxes, box)
+	}
+	slices.SortFunc(groupResources.Boxes, func(a, b logic.BoxRef) int {
+		return cmp.Compare(fmt.Sprintf("%d:%s", a.App, a.Name), fmt.Sprintf("%d:%s", b.App, b.Name))
+	})
+
+	// Sort assets holdings
+	// To sort asset holdings, we turn the asset into a string, concat with the address, and then sort
+	for holding := range groupResourceTracker.AssetHoldings {
+		groupResources.AssetHoldings = append(groupResources.AssetHoldings, holding)
+	}
+	slices.SortFunc(groupResources.AssetHoldings, func(a, b ledgercore.AccountAsset) int {
+		return cmp.Compare(fmt.Sprintf("%d:%s", a.Asset, a.Address), fmt.Sprintf("%d:%s", b.Asset, b.Address.GetUserAddress()))
+	})
+
+	// Sort app locals
+	// To sort app locals, we turn the app into a string, concat with the address, and then sort
+	for local := range groupResourceTracker.AppLocals {
+		groupResources.AppLocals = append(groupResources.AppLocals, local)
+	}
+	slices.SortFunc(groupResources.AppLocals, func(a, b ledgercore.AccountApp) int {
+		return cmp.Compare(fmt.Sprintf("%d:%s", a.App, a.Address), fmt.Sprintf("%d:%s", b.App, b.Address.GetUserAddress()))
+	})
+
+	// First populate resources that HAVE to be assigned to a specific transaction
+	for i, tracker := range txnResources {
+		// Sort assets
+		sortedAssets := make([]basics.AssetIndex, 0, len(tracker.Assets))
+		for asset := range tracker.Assets {
+			sortedAssets = append(sortedAssets, asset)
+		}
+		slices.SortFunc(sortedAssets, func(a, b basics.AssetIndex) int {
+			return cmp.Compare(a, b)
+		})
```

Comment:
```suggestion
		slices.Sort(sortedAssets)
```

etc, for below SortFunc's that sort an Ordered type.